dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -98,23 +97,9 @@ MPPTunnel::~MPPTunnel()
     LOG_FMT_TRACE(log, "destructed tunnel obj!");
 }
 
-void MPPTunnel::finishSendQueue(bool drain)
-{
-    if (tunnel_sender)
-    {
-        if (drain)
-            tunnel_sender->finishAndDrain();
-        else
-            tunnel_sender->finish();
-    }
-}
-
 /// exit abnormally, such as being cancelled.
-void MPPTunnel::close(const String & reason)
+void MPPTunnel::close(const String & reason, bool wait_sender_finish)
 {
-    SCOPE_EXIT({
-        finishSendQueue(true); // drain the send_queue when close, to release useless memory
-    });
```

Comment:
Seems `cancelWith` doesn't free the memory. Is it expected?

Reply:
Yes, since each TrackedMPPPacket holds a shared ptr of memory tracker, there is no need to drian the queue when cancel it.